spec/functions/fqdn_rotate_spec.rb

```diff
@@ -40,4 +40,21 @@ class AlsoString < String
     result = scope.function_fqdn_rotate([value])
     result.size.should(eq(4))
   end
+
+  it "should use the Puppet::Util.deterministic_rand function if available" do
+    scope.expects(:lookupvar).with("::fqdn").returns("127.0.0.1")
+    if Puppet::Util.respond_to?(:deterministic_rand)
+      Puppet::Util.expects(:deterministic_rand).with(113646079810780526294648115052177588845,4)
+    end
+    scope.function_fqdn_rotate(["asdf"])
+  end
+
+  it "should not leave the global seed in a deterministic state" do
```

Comment:
In the event that we change `fqdn_rotate()` to use `fqdn_rand()` to generate numbers to avoid duplicating `Puppet::Util.deterministic_rand`, we'll need to modify this to only execute on Puppet >=3.2.0.